code/modules/events/space_vines/vine_mutations.dm

```diff
@@ -0,0 +1,354 @@
+/datum/spacevine_mutation
+	/// Displayed name of mutation
+	var/name = ""
+	/// Severity of mutation in terms of gameplay, affects appearance chance and how many mutations can be on the same vine
+	var/severity = 1
+	var/hue
+	var/quality
+
+/datum/spacevine_mutation/proc/add_mutation_to_vinepiece(obj/structure/spacevine/holder)
+	holder.mutations |= src
+	holder.add_atom_colour(hue, FIXED_COLOUR_PRIORITY)
+
+/datum/spacevine_mutation/proc/process_mutation(obj/structure/spacevine/holder)
+	return
+
+/datum/spacevine_mutation/proc/on_birth(obj/structure/spacevine/holder)
+	return
+
+/datum/spacevine_mutation/proc/on_grow(obj/structure/spacevine/holder)
+	return
+
+/datum/spacevine_mutation/proc/on_death(obj/structure/spacevine/holder)
+	return
+
+/datum/spacevine_mutation/proc/on_hit(obj/structure/spacevine/holder, mob/hitter, obj/item/item, expected_damage)
+	. = expected_damage
+
+/datum/spacevine_mutation/proc/on_cross(obj/structure/spacevine/holder, mob/crosser)
+	return
+
+/datum/spacevine_mutation/proc/on_chem(obj/structure/spacevine/holder, datum/reagent/chem)
+	return
+
+/datum/spacevine_mutation/proc/on_eat(obj/structure/spacevine/holder, mob/living/eater)
+	return
+
+/datum/spacevine_mutation/proc/on_spread(obj/structure/spacevine/holder, turf/target)
+	return
+
+/datum/spacevine_mutation/proc/on_buckle(obj/structure/spacevine/holder, mob/living/buckled)
+	return
+
+/datum/spacevine_mutation/proc/on_explosion(severity, target, obj/structure/spacevine/holder)
+	return
+
+/datum/spacevine_mutation/proc/additional_atmos_processes(obj/structure/spacevine/holder, datum/gas_mixture/air)
+	return
+
+/datum/spacevine_mutation/aggressive_spread/proc/aggrospread_act(obj/structure/spacevine/vine, mob/living/M)
+	return
+
+/datum/spacevine_mutation/light
+	name = "Light"
+	hue = "#B2EA70"
+	quality = POSITIVE
+	severity = SEVERITY_TRIVIAL
+
+/datum/spacevine_mutation/light/on_grow(obj/structure/spacevine/holder)
+	if(holder.energy)
+		holder.set_light(LIGHT_MUTATION_BRIGHTNESS, 0.3)
+
+/datum/spacevine_mutation/toxicity
+	name = "Toxic"
+	hue = "#9B3675"
+	severity = SEVERITY_AVERAGE
+	quality = NEGATIVE
+
+/datum/spacevine_mutation/toxicity/on_cross(obj/structure/spacevine/holder, mob/living/crosser)
+	if(issilicon(crosser))
+		return
+	if(prob(TOXICITY_MUTATION_PROB) && istype(crosser) && !isvineimmune(crosser))
+		to_chat(crosser, span_alert("You accidentally touch the vine and feel a strange sensation."))
+		crosser.adjustToxLoss(20)
+
+/datum/spacevine_mutation/toxicity/on_eat(obj/structure/spacevine/holder, mob/living/eater)
+	if(!isvineimmune(eater))
+		eater.adjustToxLoss(20)
+
+/datum/spacevine_mutation/explosive  // JC IT'S A BOMB
+	name = "Explosive"
+	hue = "#D83A56"
+	quality = NEGATIVE
+	severity = SEVERITY_MAJOR
+
+/datum/spacevine_mutation/explosive/on_explosion(explosion_severity, target, obj/structure/spacevine/holder)
+	if(explosion_severity < 3)
+		qdel(holder)
+	else
+		. = 1
+		QDEL_IN(holder, 5)
```

Comment:
```suggestion
		QDEL_IN(holder, 0.5 SECONDS)
		return TRUE
```

also have this proc `return FALSE` here as well as on parent by default so we don't keep passing nulls instead of FALSE since this is clearly intended to be some sort of boolean proc (esp. with how it's consumed)